src/dvclive/dvc.py

```diff
@@ -46,3 +53,58 @@ def make_checkpoint():
+    # fmt: on
+    while True:
+        adjective = choice(ADJECTIVES)  # nosec B311
+        noun = choice(NOUNS)  # nosec B311
+        name = f"{adjective}-{noun}"
+
+        exp_ref = ExpRefInfo(baseline_rev, name)
+        try:
+            dvc_repo.experiments._validate_new_ref(exp_ref)
+        except ExperimentExistsError:
+            continue
+
+        return name
+
+
+def make_dvcyaml(live):
+    if not os.path.exists(live.dvc_file):
```

Comment:
Will `dvc.yaml` get deleted each time a new live instance is created? Maybe we need some option to configure whether to overwrite it. In some cases, I may log new stuff and want to overwrite, but in others I might have added some custom config that I want to keep.

Reply:
> Will dvc.yaml get deleted each time a new live instance is created? 

It should not be deleted and it should preserve the manual modifications from users.

> In some cases, I may log new stuff and want to overwrite, but in others I might have added some custom config that I want to keep.

What would be overwritten? The current version is extremely simple as it doesn't configure sklearn plots (added a follow-up https://github.com/iterative/dvclive/issues/371)

Reply:
Sounds good. Not a blocker.

> What would be overwritten? 

As you say, it will include sklearn plots config, which seems like the most complicated scenario, so let's discuss there.

Reply:
It should probably just write to the root of the repo, not to its own file by default. It is the dvc way, and this is against the principle of having one file to rule them all.

Reply:
🤔 But the dvclive way is to create a self-contained folder 😄 . IMHO dvclive should not write to the user's root `dvc.yaml` file, because it it makes it hard to share only the dvclive output, and it mixes machine-generated config with user-generated config. Also, if there is some conflict between the existing config and what dvclive generates, I would rather have dvclive run successfully and be able to later debug the conflicts between the files.